src/basic/pidfd-util.c

```diff
@@ -21,6 +21,26 @@
 
 static int have_pidfs = -1;
 
+int pidfd_open_safe(pid_t pid, unsigned flags) {
+        /* Unfortunately, pidfd_open() may fail with EINVAL even when passed valid arguments, which occurs
+         * when the kernel found the 'struct pid' object for the process but it does not contain any tasks.
+         * See pidfd_open() -> pidfd_create() -> pidfd_prepare() in the kernel source.
+         * That should be really fixed in the kernel, but for now let's workaround the issue by first
+         * checking the validity of the arguments, and then converting EINVAL from pidfd_open() to ESRCH. */
+
+        if (pid <= 0)
```

Comment:
If we have a wrapper anyways, maybe accept 0 as our own process like all other helpers? But feel free to ignore if you want this to stay closer to pidfd_open()

Reply:
I'd like to make this only update the spurious errno.
I think in most places we should use PidRef rather than pidfd, and pidfd_open() should be called when it is really really necessary. As you know, PidRef already gracefully handle zero pid, hence I think it is not necessary to handle zero here.